### PR TITLE
disable TestReceiveEventsAndMetadata in Windows

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

- Disable TestReceiveEventsAndMetadata on windows
- Support windows-2016 for filebeat.
- But there are two test failures -> #21757

## Why is it important?

Use windows-2016 for filebeat

## Related issues

- Relates #19641
- Relates #21757

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by jsoriano** [comment]
> But there are two test failures -> #21757

@v1v could you also skip them on Windows in this PR?

Thanks!

**Comment by v1v** [comment]
@jsoriano see https://github.com/elastic/beats/pull/21750/commits/e3084aed5c75ae4b747a080d2e807384e7392fff, although why it does not fail on `windows-2019`?  Should the skip be only for versions <= `windows-2016`? If so, how do we do that?

**Comment by jsoriano** [comment]
> although why it does not fail on `windows-2019`?

Looking at the test it seems that it fails because it cannot open a unix socket. It seems that this was something [recently added to Windows](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/), so it won't work in older versions.

>  Should the skip be only for versions <= windows-2016? If so, how do we do that?

Maybe the test should try to open a unix socket, and if it cannot, skip the test.

But feel free to skip it by now, and let this as an action point in #21757.

File: filebeat/inputsource/unix/server_test.go [hunk 1]
```diff
@@ -60,6 +60,10 @@ func TestErrorOnEmptyLineDelimiter(t *testing.T) {
 }
 
 func TestReceiveEventsAndMetadata(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("test is only supported on non-windows")
```
Comment (mdelapenya) [inline filebeat/inputsource/unix/server_test.go:64]: Do we need the same comment as in https://github.com/elastic/beats/pull/21750/files#diff-f78953ffe73980d156eb9bc06aa6cbaa4e32d81dd629d2d37af441c585934ef0R263?

**Review by mdelapenya — APPROVED** [review]

File: filebeat/inputsource/unix/server_test.go (hunk 1 above)
Comment (v1v) [reply filebeat/inputsource/unix/server_test.go:64]:
```suggestion
		t.Skip("test is only supported on non-windows. See https://github.com/elastic/beats/issues/19641")
```

**Review by jsoriano — APPROVED** [review]
Thanks!

**Pull request merged**